lib/firmata.js

```diff
@@ -1773,12 +1773,14 @@ Board.prototype.stepper2Config = function(opts) {
     data.push(opts.motorPin4);
   }
 
-  if (opts.type === this.STEPPER.TYPE.DRIVER && opts.ms1Pin && opts.ms2Pin) {
+  if ((opts.type === this.STEPPER.TYPE.DRIVER) && opts.ms1Pin && opts.ms2Pin) {
```

Comment:
Keep in mind that for some boards the pin number could be zero which would result in this evaluating to false. Same issue line 1781 as well. Could use `opts.somePin != null` (which is shorter than` typeof opts.somePin !== 'undefined'`, but some developers prefer the later and others the former) to get around this.